src/swc/aeon/io/api.py

```diff
@@ -27,7 +29,13 @@ def aeon(seconds):
     return to_datetime(seconds)  # pragma: no cover
 
 
-def to_datetime(seconds):
+@overload
+def to_datetime(seconds: float) -> datetime.datetime: ...
+@overload
+def to_datetime(seconds: pd.Series) -> pd.Series: ...
+@overload
+def to_datetime(seconds: pd.Index) -> pd.Index: ...
+def to_datetime(seconds: float | pd.Index | pd.Series) -> datetime.datetime | pd.Index | pd.Series:
```

Comment:
since the overloads already precisely define the types, do we still need this "broad union" ? 
```suggestion
def to_datetime(seconds):
```